lib/stylus-globals.js

```diff
@@ -54,17 +55,193 @@ function getCurlStatement (baseUri, method, resource) {
   const parentUrl = resource.parentUrl || ''
   const relativeUri = resource.relativeUri || '/'
 
-  const headers = method.headers
+  let headers = method.headers
     .filter((header) => header.examples && header.examples.length !== 0)
-    .map((header) => ` \\\n\t-H "${header.key}: ${header.examples[0].value}"`).join('')
+    .map((header) => ` \\\n\t-H "${header.key}: ${header.examples[0].value}"`)
 
   let params = method.queryParameters
     .filter((param) => param.examples && param.examples.length !== 0)
     .map((param) => `${param.key}=${param.examples[0].value}`)
-    .join('&')
-  params = params === '' ? '' : `?${params}`
 
-  return `curl -X ${method.method.toUpperCase()} "${baseUri}${parentUrl}${relativeUri}${params}"${headers}${payload}`
+  const resourceOptions = {
+    method: method.method,
+    baseUri: baseUri,
+    path: `${parentUrl}${relativeUri}`,
+    params: params,
+    headers: headers,
+    payload: payload
+  }
+
+  const commands = []
+  if (securitySchemes && method.securedBy) {
+    method.securedBy.forEach((methodScheme) => {
+      if (methodScheme) {
+        let securityScheme = securitySchemes[methodScheme.schemeName]
+        if (securityScheme) {
+          Array.prototype.push.apply(commands, buildCurlCommands(resourceOptions, securityScheme))
```

Comment:
wouldn't 
```javascript
commands = commands.concat(buildCurlCommands(resourceOptions, securityScheme))
```
do the same?

Reply:
If I changed `const commands` to `let commands` on line 75, yes.